### PR TITLE
JAPI-166 Expose all 7.2+ WSDFU create file options

### PR DESCRIPTION
**Pull request by rpastrana**
- Adds new methods which expose previously hidden options
- Adds wrapper around wsdfu filetype enum

Signed-off-by: Pastrana <PastraRX@RISK.REGN.NET>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu please take a look

File: wsclient/src/main/java/org/hpccsystems/ws/client/wrappers/wsdfu/DFUFileTypeWrapper.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    public static final java.lang.String _Csv = "Csv";
+    public static final DFUFileTypeWrapper Flat = new DFUFileTypeWrapper(_Flat);
+    public static final DFUFileTypeWrapper Index = new DFUFileTypeWrapper(_Index);
+    public static final DFUFileTypeWrapper Xml = new DFUFileTypeWrapper(_Xml);
+    public static final DFUFileTypeWrapper Csv = new DFUFileTypeWrapper(_Csv);
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/wrappers/wsdfu/DFUFileTypeWrapper.java]: Tabbing is off here

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java [hunk 2]
```diff
@@ -1184,12 +1185,31 @@ else if (targetVersion.major > 7 || targetVersion.major == 7 && targetVersion.mi
      * @param cluster
      * @param eclRecordDefinition
      * @param expirySeconds
+     * @param compressed
      * @return
      * @throws Exception
      */
-    public DFUCreateFileWrapper createFile(String fileName, String cluster, String eclRecordDefinition, int expirySeconds) throws Exception
+    public DFUCreateFileWrapper createFile(String fileName, String cluster, String eclRecordDefinition, int expirySeconds, Boolean compressed) throws Exception
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java]: Can we keep a signature of createFile with compression defaulting to false?

**Comment by jpmcmu** [comment]
@rpastrana looks good

**Pull request merged**